src/types/client-configuration.ts

```diff
@@ -1,10 +1,54 @@
+import { body, ValidationChain } from "express-validator";
+import { nameof } from "./util/nameof";
+import {
+  VALID_CLAIMS,
+  VALID_LOC_VALUES,
+  VALID_TOKEN_SIGNING_ALGORITHMS,
+} from "../constants";
+import { UserIdentity } from "./user-info";
+
 export default interface ClientConfiguration {
-  clientId: string;
-  publicKey: string;
-  scopes: string[];
-  redirectUrls: string[];
-  claims: string[];
-  identityVerificationSupported: boolean;
-  idTokenSigningAlgorithm: string;
-  clientLoCs: string[];
+  clientId?: string;
+  publicKey?: string;
+  scopes?: string[];
+  redirectUrls?: string[];
+  claims?: (keyof UserIdentity)[];
+  identityVerificationSupported?: boolean;
+  idTokenSigningAlgorithm?: string;
+  clientLoCs?: string[];
 }
+
+export const generateClientConfigurationPropertyValidators = (
```

Comment:
I'm not super familiar with these validators, they seem pretty readable, but some tests for these might be nice 🙂

Reply:
I've added a few examples in the config integration test. We could add some more thorough tests, tough I'm not sure what the best way to do that would be. Given the way express-validator runs as a middleware, the simplest way would be to add a bunch more test cases to the config endpoint integration test, but given it's an endpoint that's in a state of flux at the moment that might be quite brittle.

Reply:
We could come up with some tests helper function that spins up an express server with some test endpoint and pass in the validator to that. That might be a nice way to test each type individually.

Reply:
I've adde the helper function and some individual tests for validated types.

Reply:
Cool this makes more sense to me!